packages/odyssey-react-mui/src/Dialog.tsx

```diff
@@ -139,9 +140,17 @@ const Dialog = ({
   ].filter(Boolean);
 
   return (
-    <MuiDialog data-se={testId} open={isOpen} onClose={onClose}>
-      <DialogTitle translate={translate}>
-        {title}
+    <MuiDialog
+      data-se={testId}
+      open={isOpen}
+      onClose={onClose}
+      aria-labelledby="ods-dialog-title"
+    >
+      <DialogTitle
+        id={`ods-dialog-title__container-${useUniqueId()}`} // We need to explicitly unset `id` for MUI to automatically set it based on the `aria-labelledby` prop passed to `MuiDialog`
```

Comment:
You can't use hooks inline like this. It will cause render issues if we ever have a condition. It works in this case, but it's a bad practice and might fail ESLint in the future.

You can take the resulting value as a variable and pass it in no problem. But if this ever gets conditionally rendered, it will break things and only in consuming apps. Very hard bug to find.